src/main/java/org/opensearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java

```diff
@@ -333,14 +337,18 @@ public void stopAnomalyDetectorJob(String detectorId) {
             public void onResponse(StopDetectorResponse stopDetectorResponse) {
                 if (stopDetectorResponse.success()) {
                     logger.info("AD model deleted successfully for detector {}", detectorId);
-                    adTaskManager.updateLatestRealtimeTask(detectorId, ADTaskState.STOPPED, null, listener);
+                    // StopDetectorTransportAction will send out DeleteModelAction which will clear all realtime cache.
+                    // Pass null transport service to method "stopLatestRealtimeTask" to not re-clear coordinating node cache.
+                    adTaskManager.stopLatestRealtimeTask(detectorId, ADTaskState.STOPPED, null, null, listener);
```

Comment:
Why not clear here? Can you point me to the line where the cache is initially cleared for the coordinating node?

Reply:
Check line 316, when stop detector, client will send out `StopDetectorAction` , and `StopDetectorTransportAction` will send out `DeleteModelAction` which is broadcast request;  then `DeleteModelTransportAction` will clean up realtime caches for the detector.